File: C++/Sheep.cpp
#include "CONSTANTS.h"

#include "Sheep.h"

Sheep::Sheep(World &world, const Point &position) : Animal(world, position, SHEEP_SYMBOL, SHEEP_STRENGTH, SHEEP_INITIATIVE, SHEEP_COLOR, SHEEP_NAME) {}

Organism* Sheep::Clone(const Point &position) {
  return new Sheep(this->world, position);
}

File: Java/Sheep.java
public class Sheep extends Animal {
  public Sheep(World world, Point position) {
    super(world, position, Constants.SHEEP_SYMBOL, Constants.SHEEP_STRENGTH, Constants.SHEEP_INITIATIVE, Constants.SHEEP_COLOR, Constants.SHEEP_NAME, Constants.ANIMAL_RANGE);
  }

  @Override
  public final Organism Clone(Point position) {
    return new Sheep(this.world, position);
  }
}
